avocado/core/runner.py

```diff
@@ -239,7 +239,8 @@ def finish(self, proc, started, timeout, step):
                        test_state['name'])
         if proc.is_alive():
             TEST_LOG.warning("Killing hanged test process %s" % proc.pid)
-            for _ in xrange(5):     # I really want to destroy it
+            end_time = time.time() + 60
+            while time.time() < end_time:
```

Comment:
Honestly I don't know. SIGKILL is not maskable, so it should always be delivered and handled, so probably one call would suffice. On the other hand if there is a bug where the signal is not delivered then we might want to re-send it again.

The cost of re-sending the signal should be minimal as this signal can't be treated individually, so I'd prefer keeping it, but I'm open to execute it using `%10` or so. What do you think?